src/controladores/produtos.js

```diff
@@ -0,0 +1,99 @@
+const knex = require("../conexao")
+
+const cadastrarProduto = async (req, res) => {
+    const { descricao, quantidade_estoque, valor, categoria_id } = req.body
+
+    try {
+        const categoriaIdValida = await knex('categorias').where({ id: categoria_id }).first()
+
+        if (!categoriaIdValida) {
+            return res.status(404).json({ mensagem: 'A categoria_id informada não foi encontrada.' })
+        }
+
+        const produto = await knex('produtos').insert({
+            descricao,
+            quantidade_estoque,
+            valor,
+            categoria_id
+        }).returning('*')
+
+        return res.status(201).json(produto[0])
+    } catch (error) {
+        return res.status(500).json({ mensagem: error.message })
+    }
+}
+
+const editarProduto = async (req, res) => {
+    const id = req.params.id;
+    const { descricao, quantidade_estoque, valor, categoria_id } = req.body;
+
+    try {
+        const produtoExistente = await knex("produtos").where("id", id).first();
+
+        if (!produtoExistente) {
+            return res.status(404).json({ mensagem: "Produto não encontrado." });
+        }
+
```

Comment:
É importante adicionar uma validação para verificar se a categoria informada pelo usuário existe. Quando tentamos inserir uma categoria que não existe está caindo no catch, que retorna uma mensagem de erro que não está formatada.

```suggestion
const categoriaIdValida = await knex('categorias').where({ id: categoria_id }).first()

        if (!categoriaIdValida) {
            return res.status(404).json({ mensagem: 'A categoria_id informada não foi encontrada.' })
        }
```